src/containerDeployment/containerDeploymentWebviewController.ts

```diff
@@ -0,0 +1,442 @@
+        propertyName?: keyof cd.DockerConnectionProfile,
+    ): Promise<cd.ContainerDeploymentWebviewState> {
+        const erroredInputs: string[] = [];
+        const components = propertyName
+            ? [this.state.formComponents[propertyName]]
+            : Object.values(this.state.formComponents);
+
+        for (const component of components) {
+            if (!component) continue;
+
+            const prop = component.propertyName;
+
+            // Special validation for containerName, because docker commands
+            // are called for validation
+            if (prop === "containerName") {
+                const validationResult = await dockerUtils.validateContainerName(
+                    dockerConnectionProfile[prop],
+                );
+                state.isValidContainerName = validationResult !== "";
+                if (!state.isValidContainerName) {
+                    component.validation = dockerUtils.invalidContainerNameValidationResult;
+                    erroredInputs.push(prop);
+                } else {
+                    // If the container name is valid, we can reset the validation message
+                    component.validation = { isValid: true, validationMessage: "" };
+                }
+            }
+            // Special validation for port, because docker commands
+            // are called for validation
+            else if (prop === "port") {
+                const isValidPort = await this.validatePort(
+                    dockerConnectionProfile[prop]?.toString(),
+                );
+                state.isValidPortNumber = isValidPort;
+                if (!isValidPort) {
+                    component.validation = dockerUtils.invalidPortNumberValidationResult;
+                    erroredInputs.push(prop);
+                } else {
+                    component.validation = { isValid: true, validationMessage: "" };
+                }
+            }
+            // Default validation logic
+            else if (component.validate) {
+                const result = component.validate(this.state, dockerConnectionProfile[prop]);
+                component.validation = result;
+
+                if (!result.isValid) {
+                    erroredInputs.push(prop);
+                }
+            }
+        }
+        state.formErrors = erroredInputs;
+        return state;
+    }
+
+    private getDefaultConnectionProfile(): cd.DockerConnectionProfile {
+        const connection: unknown = {
+            connectionString: undefined,
+            profileName: "",
+            encrypt: "Mandatory",
+            trustServerCertificate: true,
+            server: "",
+            database: "",
+            user: "SA",
+            password: "",
+            applicationName: connectionApplicationName,
+            authenticationType: sqlAuthentication,
+            savePassword: false,
+            containerName: "",
+            version: "2025",

```

Comment:
Is there a reason why we are serializing this information in connection profile?